wp-content/themes/workingnyc/src/js/modules/programs-archive.js

```diff
@@ -160,34 +205,178 @@ export default {
+      this.updatePagination();
+    },
 
-        let pages = _this.posts.filter(page => {
-          return (page && page.show);
-        });
+    nextPage:function(event){
+      let currPage = this.posts[1].query.page;
+      this.$set(this.query, 'page', currPage+1);
+      this.updatePagination();
+    },
 
-        if (pages) {
-          let posts = pages[pages.length - 1].posts;
-          let element = document.querySelector(`[data-js='post-${posts[0].id}']`);
+    immediatePage: function(event){
+      let change = parseInt(event.target.dataset.amount);
+      this.$set(this.query, 'page', change);
+      this.updatePagination();
+    },
 
-          if (element) {
-            element.focus();
+    updateQuery: function(taxonomy, terms) {
+      return new Promise((resolve) => { // eslint-disable-line no-undef
+        this.$set(this.query, taxonomy, terms);
+        this.$set(this.query, 'page', 1);
+        // hide all of the posts
+        if(this.posts.length>1){
+          this.posts.length=1;
+        }
+        resolve();
+      })
+      .then(this.wp)
+      .catch(message => {
+      });
+    },
+
+    /**
+     * Update the pagination array
+     * The maximum page numbers to display: 6
+     */
+    setPagination: function(){
+
+      /**
+       * If number of pages is less than 7
+       *  The totalPages = [1,2,3,4,5,6]
+       *  The Pagination = 1,2,3,4,5,6
+       * else
+       *  The totalPages = [2,3,4,5]
+       *  The Pagination = firstPage,2,3,4,5..lastPage
+       */
+
+      if(this.headers.pages<=6){
+        this.firstPage = false;
+        this.lastPage = false;
+        this.totalPages = this.headers.pages;
+      }
+      else{
+        this.firstPage = true;
+        this.lastPage = true;
+
+        //If current page is between 1 and 4, the totalPages = [2,3,4,5]
+        if(this.query.page<=4){
+          this.totalPages = [2,3,4,5]
+        }
+        else{
+          /**
+           * If the current page is greater than 4, we have to show the previous and next page number
+           */ 
+          if((this.query.page!=this.totalPages[1])&&this.headers.pages-this.query.page>4){
+            let tempPages=[4];
+            let tempNumber;
+            if(this.query.page==this.totalPages[0]){
+              tempNumber = this.totalPages[0]-1;
+            }
+            else{
+              tempNumber = this.totalPages[1];
+            }
+            for(let i=0;i<4;i++){
+              tempPages[i]=i+tempNumber;
+            }
+            this.totalPages = tempPages;
+          }
+          else{
+            /**
+             * If the total pages - current page is less than or equal to 4,
+             *  we have to display the last four page numbers
+             *  we se the totalPages array with [lastpage-4,lastpage-3,lastpage-2,lastpage-1]
+             */
+            if(this.headers.pages-this.query.page<=4){
+              let tempPages=[4];
+              let bufferFlag;
```

Comment:
Minor: can you use the same variable name for `tempNumber` and `bufferFlag`? Otherwise it seems like they have two different purposes

Reply:
Yes, we can use the same name

Reply:
Can you make this change?